odoo/osv/expression.py

```diff
@@ -788,12 +788,21 @@ def push_result(query, params):
                         # rewrite condition to match records with/without lines
                         exists = 'EXISTS' if operator in NEGATIVE_TERM_OPERATORS else 'NOT EXISTS'
                         rel_alias = _generate_table_alias(alias, field.name)
-                        push_result(f"""
-                            {exists} (
-                                SELECT 1 FROM "{comodel._table}" AS "{rel_alias}"
-                                WHERE "{rel_alias}"."{inverse_field.name}" = "{alias}".id
-                            )
-                        """, [])
+                        if comodel._active_name:
+                            active = field.context.get("active_test", True)
+                            push_result(f"""
+                                {exists} (
+                                    SELECT 1 FROM "{comodel._table}" AS "{rel_alias}"
+                                    WHERE "{rel_alias}"."{inverse_field.name}" = "{alias}".id AND "{comodel._active_name}" = "{active}"'
+                                )
+                            """, [])
```

Comment:
There should be no extra condition if `field.context.get("active_test", True)` is false.
And by the way, there is a missing alias and "true" is not "spelled" correctly in SQL:
```suggestion
                        if comodel._active_name and field.context.get("active_test", True):
                            push_result(f"""
                                {exists} (
                                    SELECT 1 FROM "{comodel._table}" AS "{rel_alias}"
                                    WHERE "{rel_alias}"."{inverse_field.name}" = "{alias}".id
                                    AND "{rel_alias}"."{comodel._active_name}" = true
                                )
                            """, [])
```